libcxx/include/__tree

```diff
@@ -997,39 +1051,72 @@ public:
           __node_holder __h = __construct_node(std::forward<_Args>(__args2)...);
           __node_base_pointer __dummy;
           auto [__parent, __child] = __find_equal(__p, __dummy, __h->__get_value());
-          __node_pointer __r       = static_cast<__node_pointer>(__child);
+          __node_pointer __r       = std::__static_fancy_pointer_cast<__node_pointer>(__child);
           if (__child == nullptr) {
-            __insert_node_at(__parent, __child, static_cast<__node_base_pointer>(__h.get()));
+            __insert_node_at(__parent, __child, std::__static_fancy_pointer_cast<__node_base_pointer>(__h.get()));
             __r = __h.release();
           }
           return pair<iterator, bool>(iterator(__r), __child == nullptr);
         },
         std::forward<_Args>(__args)...);
   }
 
+  template <class... _Args>
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX26 pair<iterator, bool>
+  __emplace_hint_unique__sfinae(true_type, const_iterator __p, _Args&&... __args) {
+    return __emplace_hint_unique(__p, __args...);
+  }
+
+  template <class... _Args>
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX26 pair<iterator, bool>
+  __emplace_hint_unique__sfinae(false_type, const_iterator, _Args&&...) {
+    // This method body should never be run. It only exists to allow for compilation. See note in
+    // __insert_unique_from_orphaned_node for more information.
+    return pair<iterator, bool>(iterator(__node_pointer()), false);
+  }
+
   template <class _ValueT = _Tp, __enable_if_t<__is_tree_value_type_v<_ValueT>, int> = 0>
-  _LIBCPP_HIDE_FROM_ABI void
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX26 void
   __insert_unique_from_orphaned_node(const_iterator __p, __get_node_value_type_t<_Tp>&& __value) {
-    __emplace_hint_unique(__p, const_cast<key_type&&>(__value.first), std::move(__value.second));
+#if _LIBCPP_STD_VER >= 26
+    if (std::is_constant_evaluated() && std::is_copy_constructible_v<decltype(__value.first)>) {
+      // We create a sfinae wrapper method here, because if the __emplace_hint_unique method gets template instantiated
+      // within __insert_unique_from_orphaned_node, the code will fail to compile where the value is not
+      // copy_constructible for runtime execution as well; unless we use `if constexpr`. Given the copy-constructible
+      // code path will be a performance regression, we want to restrict it to only execute during constant evaluation
+      //, we need to delay the template instantiation
```

Comment:
```suggestion
      //, we need to delay the template instantiation.
```
Nit

Reply:
Added a few . and capitalized first letters of sentences. Rewrote some parts.

https://github.com/llvm/llvm-project/commit/69bb9d39deb48edd3445cabbc00b46f6ca2dcd4e

let me know if this is better!